plugins/core/Standard/qRANSAC_SD/qRANSAC_SD_Commands.h

```diff
@@ -0,0 +1,514 @@
+						dir.mkpath(arg);
+					}
+					outputGroupDir = dir.cleanPath(arg);
+					cmd.print(QObject::tr("\t%1 : %2").arg(OUT_GROUP_DIR, outputGroupDir));
+					outputGrouped = true;
+				}
+				else if (param == OUT_PAIR_DIR)
+				{
+				if (cmd.arguments().empty())
+				{
+					return cmd.error(QObject::tr("\nMissing parameter: Directory after \"-%1 %2\"").arg(COMMAND_RANSAC, OUT_PAIR_DIR));
+				}
+				QString arg = cmd.arguments().takeFirst();
+				QDir dir(arg);
+				if (!dir.exists())
+				{
+					cmd.print(QObject::tr("\n%1 Does not exist\tcreating path").arg(arg));
+					dir.mkpath(arg);
+				}
+				outputPairDir  = dir.cleanPath(arg);
+				cmd.print(QObject::tr("\t%1 : %2").arg(OUT_PAIR_DIR, outputPairDir));
+				outputIndividualPairs = true;
+				}
+				else if (param == OUTPUT_INDIVIDUAL_SUBCLOUDS)
+				{
+					outputIndividualClouds = true;
+				}
+				else if (param == OUTPUT_INDIVIDUAL_PRIMITIVES)
+				{
+					outputIndividualPrimitives = true;
+				}
+				else if (param == OUTPUT_INDIVIDUAL_PAIRED_CLOUD_PRIMITIVE)
+				{
+					outputIndividualPairs = true;
+				}
+				else if (param == OUTPUT_GROUPED)
+				{
+					outputGrouped = true;
+				}
+				else if (param == ENABLE_PRIMITIVE)
+				{
+					if (cmd.arguments().empty())
+					{
+						return cmd.error(QObject::tr("Missing parameter: primitive type after \"-%1 %2\"").arg(COMMAND_RANSAC, ENABLE_PRIMITIVE));
+					}
+					QString prim = cmd.arguments().takeFirst().toUpper();
+					unsigned primCount = 0;
+					while (primitiveNames.contains(prim))
+					{
+						if (prim == PRIM_PLANE)
+						{
+							params.primEnabled[qRansacSD::RPT_PLANE] = true;
+							primCount++;
+						}
+						else if (prim == PRIM_SPHERE)
+						{
+							params.primEnabled[qRansacSD::RPT_SPHERE] = true;
+							primCount++;
+						}
+						else if (prim == PRIM_CYLINDER)
+						{
+							params.primEnabled[qRansacSD::RPT_CYLINDER] = true;
+							primCount++;
+						}
+						else if (prim == PRIM_CONE)
+						{
+							params.primEnabled[qRansacSD::RPT_CONE] = true;
+							primCount++;
+						}
+						else if (prim == PRIM_TORUS)
+						{
+							params.primEnabled[qRansacSD::RPT_TORUS] = true;
+							primCount++;
+						}
+						if (cmd.arguments().empty())
+						{
+							break;
+						}
+						prim = cmd.arguments().takeFirst().toUpper();
+					}
+					if (primCount > 0)
+					{
+						if (!primitiveNames.contains(prim))
+						{
+							cmd.arguments().push_front(prim);
+						}
+					}
+					else
+					{
+						return cmd.error(QObject::tr("No valid parameter: primitive type after \"-%1 %2\"").arg(COMMAND_RANSAC, ENABLE_PRIMITIVE));
+					}
+				}
+				if (cmd.arguments().empty())
+				{
+					break;
+				}
+				param = cmd.arguments().takeFirst().toUpper();
+			}
+			if (!paramNames.contains(param))
+			{
+				cmd.arguments().push_front(param);
+			}
+		}
+		unsigned char primCount = 0;
+		for (unsigned char k = 0; k < 5; ++k)
+		{
+			primCount += (unsigned)params.primEnabled[k];
```

Comment:
static_cast?

Reply:
yeah, the folly of copy/paste from the existing code :)
I'll change both locations